aloha-core/src/main/scala/com/eharmony/aloha/semantics/compiled/plugin/csv/CsvModelRunner.scala

```diff
@@ -179,7 +184,126 @@ object OutputType extends Enumeration {
+
+            // Create a writing function.  The first function parameter is the input, the second is the model output.
+            val writingFunc: (String, String) => String = inputPosInOutput match {
+                case InputPosition.Neither => (i, o) => o
+                case InputPosition.Before => (i, o) => s"$i$outputSep$o"
+                case InputPosition.After => (i, o) => s"$o$outputSep$i"
+                case InputPosition.Both => throw new IllegalStateException("Can't output input before and after output.")
+            }
+
+            in.foreach{ line => pStream.println(writingFunc(line, predictionFn(line).map(_.toString).getOrElse(predictionMissing))) }
+        }
+        finally {
+            if (closeOut)
+                org.apache.commons.io.IOUtils.closeQuietly(out)
+        }
+    }
+}
+
+case class LoadTestOutput(
+    out: OutputStream,
+    closeOut: Boolean,
+    in: Iterator[String],
+    inputFn: (String => Any),
+    model: Model[Any, Any],
+    outputSep: String,
+    ltConf: LoadTestConfig
+) extends PredictionOutputFormat with Timing {
+    def run(): Unit = {
+        val input = correctSizedInput()
+        val inputSize = input.size
+        val models = ParVector.fill(ltConf.threads)(model)
+
+        var loops = 0
+        var pred = 0L
+        var nonEmpty = 0L
+
+        try {
+            val pStream = new PrintStream(out)
 
+            // Clean up before calculating and reporting on memory usage.
+            1 to 5 foreach { _ => System.gc() }
+
+            pStream.println(header)
+            pStream.println(report(0, 0, 0, Float.MinPositiveValue, inputSize, models.size))
+            pStream.flush()
+
+            while (loops < ltConf.loops) {
+                val (ne, intTime) = time(models.map(m => input.foldLeft(0)((s, x) => s + m(x).fold(0)(_ => 1))).sum)
+                nonEmpty += ne
+                pred += input.size * models.size
+                loops += 1
+
+                if (loops % ltConf.reportLoopMultiple == 0) {
+                    pStream.println(report(loops, pred, nonEmpty, intTime, inputSize, models.size))
+                    pStream.flush() // Flush because these might take a long time and the output is small.
+                }
+            }
+        }
+        finally {
+            if (closeOut)
+                org.apache.commons.io.IOUtils.closeQuietly(out)
+        }
+    }
+
+    private[this] def header = {
+        val t = outputSep
+        s"loop_number${t}pred_sec${t}running_pred${t}running_nonempty_pred${t}mem_used_mb${t}mem_unallocated_mb${t}unixtime_ns"
+    }
+
+    private[this] def report(loops: Int, pred: Long, nonEmpty: Long, intervalTime: Float, intervalSize: Int, models: Int): String = {
+        val time = System.nanoTime()
+        val t = outputSep
+        val predSec = intervalSize * models / intervalTime
+        val (usedMB, unallocMB) = memStats()
+        s"$loops$t$predSec$t$pred$t$nonEmpty$t$usedMB$t$unallocMB$t$time"
```

Comment:
I checked this out and it won't work with The `Seq` /  `mkString` version.  That's because the values are downcast to AnyVal and consequently, the formatting of the integral values appears as if they are doubles.  Reverting back.